Sources/FluentKit/Properties/Timestamp.swift

```diff
@@ -74,33 +71,34 @@ extension TimestampProperty: AnyProperty {
     }
     
     public func input(to input: inout DatabaseInput) {
-        input.values[self.field.key] = .bind(self.value.map(self.formatter.timestamp(from:)))
+        let timestamp = self.value.map { date in self.formatter.timestamp(from: date) } ?? Optional<Date>.none
+        input.values[self.field.key] = .bind(timestamp)
     }
 
     public func output(from output: DatabaseOutput) throws {
         self.field.inputValue = nil
         guard output.contains(self.field.key) else { return }
 
         do {
-            let string = try output.decode(self.field.key, as: Formatter.Timestamp?.self)
-            self.field.outputValue = string.flatMap(self.formatter.date(from:))
+            let date = try self.formatter.decode(at: self.field.key, from: output)
+            self.field.outputValue = date
         } catch {
             throw FluentError.invalidField(name: self.field.key.description, valueType: Value.self, error: error)
         }
     }
 
     public func encode(to encoder: Encoder) throws {
         var container = encoder.singleValueContainer()
-        try container.encode(self.wrappedValue.flatMap(self.formatter.timestamp(from:)))
+        try self.formatter.encode(date: self.value, to: &container)
```

Comment:
Why not encode to the `Encoder`. No need for the inout that way.

Reply:
I wasn't sure if it would still work correctly. We'll wait on CI and see. 😄